.github/actions/setup/action.yaml

```diff
@@ -19,10 +19,10 @@ runs:
   using: 'composite'
   steps:
     - name: 'Install pnpm'
-      uses: pnpm/action-setup@23657c8550e9be4f7ee7e9d7fe8adda6d703bb4e # v4.0.0
+      uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # v4.0.0
```

Comment:
Is there a reason we pin to hashes instead of pinning to tags?

Reply:
Commit hashes are immutable, tags are not. Pinning to a commit hash is how GitHub's security team [recommends](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) you consume third part actions

Like anything else, this ends up becoming a judgment call to balance convenience with absolute security. My personal rule of thumb is that I'm ok pinning some things to tags...

- Actions I personally maintain (or my company maintains)
- Actions GitHub itself maintains

...and the rest I pin to commits.

To get up on a soapbox: GitHub Actions are not secure and we should all be terrified of using any actions we don't write ourselves. If you want to have a bad day, read about [Codecov's breach](https://blog.gitguardian.com/codecov-supply-chain-breach/) that affected a bunch of orgs, including my last company via Codecov's supposedly safe Github Action

Reply:
Ahh -- interesting. That totally makes sense. I had forgotten that tags are not immutable.


Thank you for the explanation.